pyiron_workflow/nodes/composite.py

```diff
@@ -26,6 +27,118 @@
     from pyiron_workflow.storage import StorageInterface
 
 
+def _extract_data(item: Channel, with_values=True, with_default=True) -> dict:
+    data = {}
+    data_dict = {"default": NOT_DATA, "value": NOT_DATA, "type_hint": None}
+    if not with_values:
+        data_dict.pop("value")
+    if not with_default:
+        data_dict.pop("default")
+    for key, value in data_dict.items():
+        if getattr(item, key) is not value:
+            data[key] = getattr(item, key)
+    return data
+
+
+def _is_internal_connection(channel: Channel, workflow: Composite, io_: str) -> bool:
```

Comment:
\*sigh\*. What I would like is _all_ channels are _only ever_ connected to other channels in the same scope. Unfortunately, nodes with no parent at all at least complicate this. Anyhow, this function isn't bad, it's just that it _ought_ to be sufficient to merely use `channel.connected`. A cleaner solution would be to first go close #587, which I just opened because here you reminded me that we have this problem 😂 then come back here and not need this function at all.